core/include/engine/Vectormath.hpp

```diff
@@ -411,17 +411,18 @@ namespace Engine
             auto& n_cell_atoms_old = geometry_old.n_cell_atoms;
 
             auto& n_cells_new = geometry_new.n_cells;
+            const int n_cells_test[] = {geometry_new.n_cells[0],geometry_new.n_cells[1],geometry_new.n_cells[2]};
```

Comment:
Will try that later

Reply:
Even when I use the `const Data::Geometry & geometry_new` from the function call to limit the loop:
`for (int i=0; i<geometry_new.n_cells_new[0]; ++i)`
the error still persists -> local copy (and maybe compilerbug-ticket) seems to be the way to go, till the underlying cause is fixed, because this is a show-stopper on Intel compiler currently.

Reply:
That is really quite weird... but what if you don't even use a reference (`const std::vector<int> n_cells_new = geometry_new.n_cells;`)?

If that does not work either, please add a comment to the line to indicate it is only needed because of the intel compiler, and I will merge it.

Reply:
Just checked vs the new 18.0.3.222 compiler. Problem still persists.

Also tried your suggestion with both intel 18 and 18.0.3.222:
----------------------------------------------------------------------------
core/include/engine/Vectormath.hpp(421): error: parallel loops with collapse must be perfectly nested
              for (int i=0; i<n_cells_new[0]; ++i)
                       ^
          detected during instantiation of "field<T> Engine::Vectormath::change_dimensions(field<T> &, const Data::Geometry &, const Data::Geometry &, T, std::array<int, 3UL>) [with T=Vector3]" at line 23 of "core/src/Spirit/Geometry.cpp"

Error: the OpenMP "single" pragma must not be enclosed by the "for" pragma
          detected during instantiation of "field<T> Engine::Vectormath::change_dimensions(field<T> &, const Data::Geometry &, const Data::Geometry &, T, std::array<int, 3UL>) [with T=Vector3]" at line 23 of "core/src/Spirit/Geometry.cpp"
-------------------------------------------------------------------------

Will add the comment and push it to here shortly.